<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <meta name="author" content="Ben Murrell">
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>clusterpipeline - RAD.jl</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="../css/highlight.css">
  <link href="../assets/Documenter.css" rel="stylesheet">
  
  <script>
    // Current page data
    var mkdocs_page_name = "clusterpipeline";
    var mkdocs_page_input_path = "clusterpipeline.md";
    var mkdocs_page_url = "/clusterpipeline/";
  </script>
  
  <script src="../js/jquery-2.1.1.min.js"></script>
  <script src="../js/modernizr-2.8.3.min.js"></script>
  <script type="text/javascript" src="../js/highlight.pack.js"></script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> RAD.jl</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="current">
	  
          
            <li class="toctree-l1">
		
    <a class="" href="..">Home</a>
	    </li>
          
            <li class="toctree-l1 current">
		
    <a class="current" href="./">clusterpipeline</a>
    <ul class="subnav">
            
    <li class="toctree-l2"><a href="#primary-clustering-and-pipeline-functions">Primary Clustering and Pipeline Functions</a></li>
    

    <li class="toctree-l2"><a href="#arguments">Arguments</a></li>
    

    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../consensus/">consensus</a>
	    </li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">RAD.jl</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
    
    <li>clusterpipeline</li>
    <li class="wy-breadcrumbs-aside">
      
        <a href="https://github.com/MurrellGroup/RAD.jl/edit/master/docs/clusterpipeline.md"
          class="icon icon-github"> Edit on GitHub</a>
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <p><a id='Primary-Clustering-and-Pipeline-Functions-1'></a></p>
<h1 id="primary-clustering-and-pipeline-functions">Primary Clustering and Pipeline Functions</h1>
<ul>
<li><a href="./#RAD.clusterpipeline"><code>RAD.clusterpipeline</code></a></li>
<li><a href="./#RAD.denoise"><code>RAD.denoise</code></a></li>
<li><a href="./#RAD.get_ave_dists"><code>RAD.get_ave_dists</code></a></li>
<li><a href="./#RAD.kmer_split_clustering"><code>RAD.kmer_split_clustering</code></a></li>
<li><a href="./#RAD.kmer_split_consensus"><code>RAD.kmer_split_consensus</code></a></li>
</ul>
<p><a id='RAD.denoise' href='#RAD.denoise'>#</a>
<strong><code>RAD.denoise</code></strong> &mdash; <em>Function</em>.</p>
<pre><code>denoise(seqs; &lt;keyword arguments&gt; )

                rough_radius::Float64 = 0.01,
                fine_radius::Float64 = 1.0,
                reassign::Bool = true,
                clust_multiplier::Float64 = 1.0,
                min_devs::Int64 = 6,
                initial_devs::Int64 = 20,
                user_min_clust::Int64 = 5,
                k::Int64 = 6,
                distfunc = euclidean,
                center = mean,
                verbose::Int64 = 1,
                cycle_lim = 20,
                triangle::Bool = false,
                degap_param::Bool = false,
                Polish::Bool = true,
                p_value::Float64 = 0.01
</code></pre>

<h1 id="arguments">Arguments</h1>
<ul>
<li><code>seqs</code>: array containing the sequences to be operated on</li>
<li><code>rough_radius::Float64</code>: distance between SNPs used in fine clustering</li>
<li><code>fine_radius::Float64</code>: distance between SNPs used in rough clustering</li>
<li><code>reassign::Bool = true</code>: indicates whether or not to reassign the rough singletons</li>
</ul>
<p>back to consensus sequences</p>
<ul>
<li><code>clust_multiplier::Float64 = 1.0</code>: minimum non-error cluster size</li>
<li><code>min_devs::Int64 = 6</code>: number of high standard deviation kmer to fine cluster on</li>
<li><code>initial_devs::Int64 = 20</code>: initial amount of high standard deviation kmer to test</li>
<li><code>user_min_clust::Int64 = 5</code>: minimum size of a cluster, competes with clust_multiplier</li>
<li><code>k::Int64 = 6</code>: length of kmer sequences calculated</li>
<li><code>distfunc = euclidean</code>: type of distance function calculated on the kmer sequences</li>
<li><code>center = mean</code>: sets the center of a centroid to the mean of the centroid</li>
<li><code>verbose::Int64 = 1</code>: when true, the status of the program will be printed to 'stdout'</li>
<li><code>cycle_lim = 20</code>: maximum number of cycles allowed in clustering call</li>
<li><code>triangle::Bool = false</code>: controls whether or not the triangle inequality is used in calculations</li>
<li><code>degap_param::Bool = false</code>: controls whether degap is used in consensus calculations</li>
<li><code>Polish::Bool = false</code>: if true, draws a consensus again from singletons after reassignment</li>
<li><code>p_value::Float64 = 0.01</code>: statistical threshold for splitting decisions</li>
</ul>
<p>Performs clustering and sequence consensus.</p>
<p>Order of process:</p>
<ul>
<li>Rough clustering</li>
<li>Fine clustering</li>
<li>Consensus of clusters</li>
</ul>
<p>Recommended <code>radius</code> for rough clustering: 0.01 Recommended <code>radius</code> for fine clustering: 1.0</p>
<p><a target='_blank' href='https://github.com/MurrellGroup/RAD.jl/blob/6151ae06c78da4f896cc88d797fefcac436053eb/src/clusterpipeline.jl#L3-L54' class='documenter-source'>source</a><br></p>
<p><a id='RAD.kmer_split_clustering' href='#RAD.kmer_split_clustering'>#</a>
<strong><code>RAD.kmer_split_clustering</code></strong> &mdash; <em>Function</em>.</p>
<pre><code>kmer_split_clustering(kmer_vecs, seqs; rough_radius::Float64 = 0.01, fine_radius::Float64 = 1.0,  reassign::Bool = true,
                    clust_multiplier::Float64 = 1.0, min_devs::Int64 = 6, initial_devs::Int64 = 20,
                    user_min_clust::Int64 = 5, k::Int64 = 6, distfunc = euclidean, center = mean,
                    verbose::Int64 = 1, cycle_lim = 20, triangle::Bool = false, p_value::Float64 = 0.01)
</code></pre>

<p>Returns the fine clustering result from the given sequences.</p>
<p><a target='_blank' href='https://github.com/MurrellGroup/RAD.jl/blob/6151ae06c78da4f896cc88d797fefcac436053eb/src/clusterpipeline.jl#L72-L80' class='documenter-source'>source</a><br></p>
<p><a id='RAD.kmer_split_consensus' href='#RAD.kmer_split_consensus'>#</a>
<strong><code>RAD.kmer_split_consensus</code></strong> &mdash; <em>Function</em>.</p>
<pre><code>kmer_split_consensus(kmer_vecs, seqs, fine_indices; verbose::Int64=1, distfunc=euclidean, reassign::Bool=true,
                   k::Int64=6, degap_param::Bool=false, user_min_clust::Int64=5, Polish::Bool=true)
</code></pre>

<p>Returns the consensus of the sequences after clustering.</p>
<p><a target='_blank' href='https://github.com/MurrellGroup/RAD.jl/blob/6151ae06c78da4f896cc88d797fefcac436053eb/src/clusterpipeline.jl#L93-L99' class='documenter-source'>source</a><br></p>
<p><a id='RAD.get_ave_dists' href='#RAD.get_ave_dists'>#</a>
<strong><code>RAD.get_ave_dists</code></strong> &mdash; <em>Function</em>.</p>
<pre><code>get_ave_dists(fine_indices, original_seqs, consensus_seqs; distfunc=euclidean, center=median, k::Int64=6)
</code></pre>

<p>Finds the average distance using the declared distance function</p>
<p><a target='_blank' href='https://github.com/MurrellGroup/RAD.jl/blob/6151ae06c78da4f896cc88d797fefcac436053eb/src/clusterpipeline.jl#L148-L152' class='documenter-source'>source</a><br></p>
<p><a id='RAD.clusterpipeline' href='#RAD.clusterpipeline'>#</a>
<strong><code>RAD.clusterpipeline</code></strong> &mdash; <em>Function</em>.</p>
<pre><code>clusterpipeline(infile::String; &lt;Keyword Arguments&gt; )

                     prefix = &quot;&quot;,
                     outfile = &quot;&quot;,
                     ref_file = &quot;&quot;,
                     error_filter = 0.01,
                     homopoly_filter = true,
                     minlength = 20,
                     maxlength = 1000000,
                     trim_seqs = true,
                     align_ref_frames = true,
                     write_bad_seqs = false,
                     write_aminoacids = true,
                     minsize = 5,
                     cycle_lim = 10,
                     merge_differing_homopolymers = false,
                     cluster_filedump_path = &quot;&quot;,
                     post_filedump = false,
                     rough_radius::Float64=0.01,
                     fine_radius::Float64=1.0,
                     reassign::Bool=true,
                     clust_multiplier::Float64=1.0,
                     min_devs::Int64=6,
                     initial_devs::Int64=20,
                     user_min_clust::Int64=5,
                     k::Int64=6,
                     distfunc=euclidean,
                     center=mean,
                     verbose::Int64=1,
                     triangle::Bool=false,
                     degap_param::Bool=false,
                     Polish::Bool=true,
                     p_value::Float64=0.01)
</code></pre>

<p><strong>Arguments</strong></p>
<ul>
<li><code>prefix = ""</code>: prefix for names of sequences in output file (ie in case of multiple time points, etc.).</li>
<li><code>outfile = ""</code>: default is "[infile name without extension]-clusters.fasta".</li>
<li><code>ref_file = ""</code>: panel of reference seqs â€“ if provided, used to orient and trim nearest sequences in input file.</li>
<li><code>error_filter = 0.01</code>: if the input is a ".fastq" file, any sequence with a site with error greater than this value is removed before clustering.</li>
<li><code>homopoly_filter = true</code>: if true, remove sequences with erroneous run-on homopolymer regions (or trim the sequences if this region is on an end) via hidden markov model inference.</li>
<li><code>minlength = 20</code>: removes sequences shorter than this prior to clustering.</li>
<li><code>maxlength = 1000000</code>: removes sequences longer than this prior to clustering.</li>
<li><code>trim_seqs = true</code>: if a reference panel is provided (<code>ref_file</code>) and this is true, locally trims consensus seqs to nearest reference sequence</li>
<li><code>align_ref_frames = true</code>: corrects codon reading frames of inferred consensus sequences.</li>
<li><code>write_bad_seqs = false</code>: write sequences with uncorrectable reading frames to file</li>
<li><code>write_aminoacids = true</code>: write amino acid translations of consensus sequences to file</li>
<li><code>minsize = 5</code>: throw away clusters/consensuses with fewer than this many reads.</li>
<li><code>cycle_lim = 10</code> : TODO</li>
<li><code>merge_differing_homopolymers = false</code>: aggregates clusters with consensus seqs that only differ by a single gap in a homopolymer region.</li>
<li><code>cluster_filedump_path = ""</code>: if nonempty, writes original sequences to output files determined by resulting cluster in given directory. This string may also include a file name prefix.</li>
<li><code>post_filedump</code>: writes inferred consensus seqs to file after clustering, before aligning and trimming.</li>
<li><code>rough_radius::Float64</code>: distance between SNPs used in fine clustering</li>
<li><code>fine_radius::Float64</code>: distance between SNPs used in rough clustering</li>
<li><code>reassign::Bool = true</code>: indicates whether or not to reassign the rough singletons</li>
</ul>
<p>back to consensus sequences</p>
<ul>
<li><code>clust_multiplier::Float64 = 1.0</code>: minimum non-error cluster size</li>
<li><code>min_devs::Int64 = 6</code>: number of high standard deviation kmer to fine cluster on</li>
<li><code>initial_devs::Int64 = 20</code>: initial amount of high standard deviation kmer to test</li>
<li><code>user_min_clust::Int64 = 5</code>: minimum size of a cluster, competes with clust_multiplier</li>
<li><code>k::Int64 = 6</code>: length of kmer sequences calculated</li>
<li><code>distfunc = euclidean</code>: type of distance function calculated on the kmer sequences</li>
<li><code>center = mean</code>: sets the center of a centroid to the mean of the centroid</li>
<li><code>verbose::Int64 = 1</code>: when true, the status of the program will be printed to 'stdout'</li>
<li><code>cycle_lim = 20</code>: maximum number of cycles allowed in clustering call</li>
<li><code>triangle::Bool = false</code>: controls whether or not the triangle inequality is used in calculations</li>
<li><code>degap_param::Bool = false</code>: controls whether degap is used in consensus calculations</li>
<li><code>Polish::Bool = false</code>: if true, draws a consensus again from singletons after reassignment</li>
<li><code>p_value::Float64 = 0.01</code>: statistical threshold for splitting decisions</li>
</ul>
<p>A clustering pipeline for sequence files. Order of processes:</p>
<ul>
<li>Error filter (if .fastq)</li>
<li>Length filter</li>
<li>Homopolymer filter</li>
<li>Length filter again</li>
<li>Orient (either to reference panel or inferred coarse cluster centroid)</li>
<li>Rough clustering</li>
<li>Fine clustering</li>
<li>Consensus of clusters</li>
</ul>
<p>Recommended <code>radius</code> for rough clustering: 0.01 Recommended <code>radius</code> for fine clustering: 1.0</p>
<p><code>infile</code> and <code>ref_file</code> must be either .fasta or .fastq. If file is .fastq, then removes sequences with greater than <code>error_filter</code> error.</p>
<p><a target='_blank' href='https://github.com/MurrellGroup/RAD.jl/blob/6151ae06c78da4f896cc88d797fefcac436053eb/src/clusterpipeline.jl#L166-L257' class='documenter-source'>source</a><br></p>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../consensus/" class="btn btn-neutral float-right" title="consensus">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href=".." class="btn btn-neutral" title="Home"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
          <a href="https://github.com/MurrellGroup/RAD.jl/" class="fa fa-github" style="float: left; color: #fcfcfc"> GitHub</a>
      
      
        <span><a href=".." style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="../consensus/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS_HTML"></script>
      <script src="../assets/mathjaxhelper.js"></script>
      <script src="../search/require.js"></script>
      <script src="../search/search.js"></script>

</body>
</html>
